### PR TITLE
DOCS-12368-dw-object-sample-kt

### PR DESCRIPTION
**Pull request by kevintroller**
Adding new DW example

### CODE REVIEW

File: modules/ROOT/pages/_partials/cookbook-dw/map-an-object-ex02/transform.dwl [hunk 1]
```diff
@@ -0,0 +1,13 @@
+%dw 2.0
+import * from dw::core::Objects
+output application/csv
+---
+valueSet(payload groupBy ((item, index) -> (item.FirstName ++ item.LastName ++ item.Age)) mapObject ((value, key, index) -> (index): {
```
Comment (svacas) [inline modules/ROOT/pages/_partials/cookbook-dw/map-an-object-ex02/transform.dwl:5]: `valueSet` is deprecated since 2.3, use `valuesOf` instead

**Review by svacas — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/_partials/cookbook-dw/map-an-object-ex02/transform.dwl [hunk 2]
```diff
@@ -0,0 +1,13 @@
+%dw 2.0
```
Comment (TomoDC) [inline modules/ROOT/pages/_partials/cookbook-dw/map-an-object-ex02/transform.dwl:1]: Maybe formatting the script:

```
%dw 2.0
output application/csv
---
valuesOf(
    payload 
        groupBy ((item, index) -> (
            item.FirstName ++ item.LastName ++ item.Age)) 
        mapObject ((value, key, index) -> 
            (index): {
                "Sr.No.": value."Sr.No." joinBy ":", 
                "FirstName": value.FirstName[0],
                "LastName": value.LastName[0],
                "Age": value.Age[0],
                "Team Name": value."Team Name" joinBy ":", 
                "Role": value.Role joinBy ":"
            })
)
```

File: modules/ROOT/pages/dataweave-cookbook-map-an-object.adoc [hunk 3]
```diff
@@ -34,6 +37,32 @@ include::partial$cookbook-dw/map-an-object/inputs/payload.json[]
 include::partial$cookbook-dw/map-an-object/out.json[]
 ----
 
+== Second Example
+
+This example uses the `mapObject` function to iterate through the elements in a table. If two rows have the same values in the `FirstName`, `LastName` or `Age` columns, the DataWeave script transforms both rows into a single row in a CSV file. The remaining values in the `Team Name` and `Role` columns in the row file are concatenated with `:` in the single CSV row.
```
Comment (TomoDC) [inline modules/ROOT/pages/dataweave-cookbook-map-an-object.adoc:42]:
```suggestion
This example uses the `mapObject` function to iterate through the keys and values of the object that results from using `groupBy` on the payload. If some objects of the input payload have the same values in the `FirstName`, `LastName` and `Age` keys, the DataWeave script transforms those objects into a single row in a CSV file. The remaining values in the `Team Name` and `Role` keys for those objects are concatenated with `:` in the single CSV row.
```
Transform from json objects to the CSV file

File: modules/ROOT/pages/dataweave-cookbook-map-an-object.adoc [hunk 4]
```diff
@@ -4,11 +4,14 @@ include::_attributes.adoc[]
 endif::[]
 :keywords: studio, anypoint, transform, transformer, format, aggregate, rename, split, filter convert, json, metadata, dataweave, data weave, datamapper, dwl, dfl, dw, output structure, input structure, map, mapping, as
 
-
-This DataWeave example uses both the `map` and `mapObject` functions to iterate through the input and set all of the keys to upper case.
+The following DataWeave examples use the `mapObject` function to iterate through the keys and values in each of the objects of the arrays.
```
Comment (TomoDC) [inline modules/ROOT/pages/dataweave-cookbook-map-an-object.adoc:7]:
```suggestion
The following DataWeave examples use the `mapObject` function to iterate through the keys and values of objects.
```
It does not iterate over arrays. The array are transformed by other functions to an object and the mapObject iterate through that object.

File: modules/ROOT/pages/dataweave-cookbook-map-an-object.adoc [hunk 5]
```diff
@@ -34,6 +37,32 @@ include::partial$cookbook-dw/map-an-object/inputs/payload.json[]
...
+----
+In the previous input example, `Sr.No. 2` and `Sr.No. 3`, and also `Sr.No. 5` and `Sr.No. 6` are exactly the same person because they have same first name, last name and age. However, `Sr.No. 4` is different person with same first and last name but different age.
+
+.DataWeave Script:
+[source,dataweave,linenums]
+----
+include::partial$cookbook-dw/map-an-object-ex02/transform.dwl[]
+----
+
+The DataWeave scripts merges the repeated values of columns `FirstName`, `LastName` or `Age` of rows `Sr.No. 2` and `Sr.No. 3`, and also rows `Sr.No. 5` and `Sr.No. 6`. These values are separated with `,` in the sigle CSV row. The values of the `Team Name` and `Role` columns are concatenated with `:` in the sigle CSV row.
```
Comment (TomoDC) [inline modules/ROOT/pages/dataweave-cookbook-map-an-object.adoc:59]:
```suggestion
The DataWeave script merges the repeated values of keys `FirstName`, `LastName` and `Age` of the objects with key-value `Sr.No. 2` and `Sr.No. 3`, and also `Sr.No. 5` and `Sr.No. 6`. These values are separated with `,` in the single CSV row. The values of the `Team Name` and `Role` for those objects are concatenated with `:` in the single CSV row.
```
From JSON objects to CSV rows.

File: modules/ROOT/pages/dataweave-cookbook-map-an-object.adoc [hunk 6]
```diff
@@ -34,6 +37,32 @@ include::partial$cookbook-dw/map-an-object/inputs/payload.json[]
...
+----
+include::partial$cookbook-dw/map-an-object-ex02/transform.dwl[]
+----
+
+The DataWeave script merges the repeated values of keys `FirstName`, `LastName` and `Age` of the objects with key-value `Sr.No. 2` and `Sr.No. 3`, and also `Sr.No. 5` and `Sr.No. 6`. These values are separated with `,` in the single CSV row. The values of the `Team Name` and `Role` for those objects are concatenated with `:` in the single CSV row.
+
+.Output CSV:
+[source,CSV,linenums]
+----
+include::partial$cookbook-dw/map-an-object-ex02/out.json[]
```
Comment (svacas) [inline modules/ROOT/pages/dataweave-cookbook-map-an-object.adoc:64]: change the file extension from `out.json` to `out.csv`

**Review by svacas — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/dataweave-cookbook-map-an-object.adoc [hunk 7]
```diff
@@ -34,6 +37,32 @@ include::partial$cookbook-dw/map-an-object/inputs/payload.json[]
...
+This example uses the `mapObject` function to iterate through the keys and values of the object that results from using `groupBy` on the payload. If some objects of the input payload have the same values in the `FirstName`, `LastName` and `Age` keys, the DataWeave script transforms those objects into a single row in a CSV file. The remaining values in the `Team Name` and `Role` keys for those objects are concatenated with `:` in the single CSV row.
+
+[[ex02]]
+
+.Input JSON Payload:
+[source,json,linenums]
+----
+include::partial$cookbook-dw/map-an-object-ex02/inputs/payload.json[]
+----
+In the previous input example, `Sr.No. 2` and `Sr.No. 3`, and also `Sr.No. 5` and `Sr.No. 6` are exactly the same person because they have same first name, last name and age. However, `Sr.No. 4` is different person with same first and last name but different age.
```
Comment (luanamulesoft) [inline modules/ROOT/pages/dataweave-cookbook-map-an-object.adoc:51]: Suggestion: "In the previous input example, the object pair `Sr.No. 2` and `Sr.No. 3`, as well as the object pair `Sr.No. 5` and `Sr.No. 6`, refer to the same person, as they have the same first name, last name, and age. However, `Sr.No. 4` refers to a different person, as it has the same first and last name but a different age.

File: modules/ROOT/pages/dataweave-cookbook-map-an-object.adoc [hunk 8]
```diff
@@ -34,6 +37,32 @@ include::partial$cookbook-dw/map-an-object/inputs/payload.json[]
...
+
+.DataWeave Script:
+[source,dataweave,linenums]
+----
+include::partial$cookbook-dw/map-an-object-ex02/transform.dwl[]
+----
+
+The DataWeave script merges the repeated values of keys `FirstName`, `LastName` and `Age` of the objects with key-value `Sr.No. 2` and `Sr.No. 3`, and also `Sr.No. 5` and `Sr.No. 6`. These values are separated with `,` in the single CSV row. The values of the `Team Name` and `Role` for those objects are concatenated with `:` in the single CSV row.
```
Comment (luanamulesoft) [inline modules/ROOT/pages/dataweave-cookbook-map-an-object.adoc:59]: Suggestion: The DataWeave script merges the repeated values of `FirstName`, `LastName` and `Age` of the objects with key-value `Sr.No. 2` and `Sr.No. 3`, as well as `Sr.No. 5` and `Sr.No. 6`. These values are separated with `,` in the single CSV row. The values of `Team Name` and `Role` for those objects are concatenated with `:` in the single CSV row.

**Review by luanamulesoft — DISMISSED** [review]

**Review by luanamulesoft — APPROVED** [review]

**Review by svacas — APPROVED** [review]

**Comment by kevintroller** [comment]
Closing as creating new PR (verified commits) https://github.com/mulesoft/docs-dataweave/pull/109

**Pull request closed without merging**